ansible/roles/pki/files/usr/local/lib/pki/pki-realm

```diff
@@ -1187,6 +1188,37 @@ request_acme_tiny_certificate () {
 
 }
 
+request_acme_dns_ovh_certificate() {
+    # We expect the API secrets to be in ./private/dns-ovh-credentials.key
+    local credentials=/etc/pki/realms/${config['name']}/private/dns-ovh-credentials.key
+    local letsencrypt=/etc/letsencrypt/live/${config['pki_default_fqdn']}
+    local pkirealm=/etc/pki/realms/${config['pki_default_domain']}
+    local email=""
+
+    if [ -n "${config['acme_contacts']}" ]; then
+        email="-m${config['acme_contacts']}"
+    fi
+ 
+    certbot certonly -n --dns-ovh --agree-tos "${email}" --dns-ovh-credentials ${credentials} -d${config['pki_default_fqdn']}
```

Comment:
Yes, you can specify multiple domains with -d.

In the case of OVH, if you specify foobar.example.com and *.example.com together, it will result in a error stating that foobar is already part of the wildcard

Reply:
No, I mean if it can be done using current `pki` role variables, since the `pki_default_fqdn` is global and based on the host's FQDN. You would probably have to gather domains from the realm configuration and merge them into one string separate by commas, then pass it to the `-d` parameter.

Dealing with wildcards could be done in the script as well, but we can do that later, when basic implementation works. We could filter out any subdomains if a wildcard is requested.

Reply:
Basic implementation works, and does not yet deal with wildcard filter.

Reply:
Current implementation now handles gathering the multiple domains then filtering out non-wildcards if needed and passing it along to the certbot.